pandas/core/groupby/groupby.py

```diff
@@ -2630,6 +2630,69 @@ def _reindex_output(
 
         return output.reset_index(drop=True)
 
+    def sample(
+        self,
+        n: Optional[int] = None,
+        frac: Optional[float] = None,
+        replace: bool = False,
+        weights=None,
+        random_state=None,
+    ):
+        """
+        Return a random sample of items from each group.
```

Comment:
How much of this docstring can we share with the base implementation?

Reply:
Some of the arguments have almost identical descriptions (I made some nitty changes) while others are more specific to this implementation (e.g., weights since the normalization behavior is going to be different)

Reply:
Okay. We can create a follow up issue to potentially share the docstrings somehow